### ISSUE TITLE
stable connection file

### ISSUE THREAD

**Issue by behrisch** [eclipse/sumo#7584]
In order to make a connection patch file stable even when bike or foot paths are added later an additional syntax could be invented. Proposal:
```
<connection from="-786367651" to="51812052" fromLane="passenger:0" toLane="passenger:0"/>
```
So instead of giving the lane index only, one gives the index in the list of lanes allowing a specific vehicle class. so the above example would denote the rightmost lane allowing passenger cars.

This could also influence the syntax decision for #5443

**Comment by namdre** [comment]
The syntax looks useful.
However, there are mechanisms for adding bicycle and pedestrian lanes that preserve the original connections (attributes sidewalkWidth, bikelaneWidth). Could you use those instead?

**Comment by behrisch** [comment]
Yes in most cases this will do the trick. The use case was an OSM network which was exported without pedestrian infrastructure, then some connections were fixed and then it was exported again with pedestrians and we wanted to reuse the fixes.
